src/modules/dashboard/components/TaskEditDialog/TaskEditDialog.tsx

```diff
@@ -212,15 +216,31 @@ const TaskEditDialog = ({
           ),
         );
 
+      const refetchSelfTasks = [
+        {
+          query: UserTasksDocument,
+          variables: {
+            address: walletAddress,
+          } as UserTasksQueryVariables,
+        },
+      ];
```

Comment:
In theory, this could be done as a `cacheUpdates` function, but would be quite messy, as it would require us to expand some of our existing queries unnecessarily, just for when this case occurs (user assigns or unassigns themselves). I can't imagine this is going to happen often enough to warrant such implementation.